docs/development/zlib.md

```diff
@@ -0,0 +1,243 @@
+
+* [AutotoolsDeps](https://docs.conan.io/en/latest/reference/conanfile/tools/gnu/autotoolsdeps.html): generates the **conanautotoolsdeps.sh/bat** script with information about where the **zlib** library and its dependencies  are installed together with other information like version, flags, and directory data or configuration. This is done setting some enviroment variables for Autotools like: ``LIBS``, ``CPPFLAGS``,``CXXFLAGS``, ``CFLAGS`` and ``LDFLAGS``.  It will also generate a ``deactivate_conanautotoolsdeps.sh/bat`` so you can restore your environment.
+
+Declare these generators in your **conanfile.txt** along with your **zlib** dependency like:
+
+```ini
+[requires]
+zlib/1.2.12
+
+[generators]
+AutotoolsToolchain
+AutotoolsDeps
+```
+
+<br>
+
+Then, building your project is as easy as:
+
+```shell
+$ conan install .
+# set the environment variables for Autotools
+$ source conanautotoolstoolchain.sh
+$ source conanautotoolsdeps.sh
+$ ./configure
+$ make
+
+# restore the environment after the build is completed
+$ source deactivate_conanautotoolstoolchain.sh
+$ source deactivate_conanautotoolsdeps.sh
+```
+
+<br>
+
+### [pkg-config Conan generator](https://docs.conan.io/en/latest/reference/conanfile/tools/gnu/pkgconfigdeps.html)
+
+<br>
+
+* [PkgConfigDeps](https://docs.conan.io/en/latest/reference/conanfile/tools/gnu/pkgconfigdeps.html): generates the **zlib.pc** file (and the ones corresponding to **zlib** dependencies) with information about the dependencies that can be later used by the **pkg-config** tool pkg-config to collect data about the libraries Conan installed.
+
+<br>
+
+You can use this generator instead of the **AutotoolsDeps** one:
+
+```ini
+[requires]
+zlib/1.2.12
+
+[generators]
+AutotoolsToolchain
+PkgConfigDeps
+```
+
+<br>
+
+And then using **pkg-config** to set the environment variables you want, like:
+
+```shell
+$ conan install .
+# set the environment variables for Autotools
+$ source conanautotoolstoolchain.sh
+
+# You will have to set the PKG_CONFIG_PATH to where Conan created the zlib.pc file
+$ export CPPFLAGS="$CPPFLAGS $(pkg-config --cflags zlib)"
+$ export LIBS="$LIBS $(pkg-config --libs-only-l zlib)"
+$ export LDFLAGS="$LDFLAGS $(pkg-config --libs-only-L --libs-only-other zlib)"
+
+$ ./configure
+$ make
+
+# restore the environment after the build is completed
+$ source deactivate_conanautotoolstoolchain.sh
+```
+
+
+
+
+<br>
+
+## Other build systems
+
+<br>
+
+Please, [check the Conan documentation](https://docs.conan.io/en/latest/reference/conanfile/tools.html) for other integrations besides the ones listed in this document.
+
+
+
+
+
+
+<br>
+
+## Exposed header files for zlib
+
+<br>
+
+```cpp
+#include <zconf.h>
+#include <zlib.h>
+```
+
+<br>
+
+
+---
+---
+Conan **1.47.0**. JFrog LTD. [https://conan.io](https://conan.io). Autogenerated 2022-04-26 15:36:12.
```

Comment:
Is this file needed? I see it is autogenerated.

Reply:
wrong PR @Joeydelarago you should review the 5.1